src/gpio.rs

```diff
@@ -123,6 +123,12 @@ pub trait TouchPin: Pin {
     fn touch_channel(&self) -> touch_pad_t;
 }
 
+/// A marker trait designating a pin which is capable of
```

Comment:
This marker trait is ESP32 specific, so it has to be enabled only when compiling for the original ESP32 chip with `#[cfg(esp32)]. Also everywhere below.

Reply:
In fact, it also needs to be protected not to compile when the `riscv-ulp-hal` feature is enabled. So:
```
#[cfg(esp32)]
#[cfg(not(feature = "riscv-ulp-hal"))]
```

Reply:
moved to esp32 chip and added the feature guard